pkg/api/api.go

```diff
@@ -80,6 +88,35 @@ func (s *server) getOrCreateTag(tagUid string) (*tags.Tag, bool, error) {
 	return t, false, err
 }
 
+var errInvalidChunkAddress = errors.New("invalid chunk address")
+var errNoResolver = errors.New("no resolver connected")
+
+func (s *server) resolveNameOrAddress(str string) (swarm.Address, error) {
+	log := s.Logger
+
+	// Try and parse the name as a bzz address.
+	addr, err := swarm.ParseHexAddress(str)
+	if err == nil {
+		log.Tracef("name resolve: valid bzz address %q", str)
+		return addr, nil
+	}
+
+	// If no resolver is not available, return an error.
+	if s.Resolver == nil {
+		return swarm.ZeroAddress, errNoResolver
+	}
+
+	// Try and resolve the name using the provided resolver.
+	log.Debugf("name resolve: attempting to resolve %s to bzz address", str)
+	addr, err = s.Resolver.Resolve(str)
+	if err == nil && !addr.IsZero() {
```

Comment:
try to keep happy path non-branching. if the error is not nil, then return the error, otherwise return the address. also, if the error is nil, the address should not be the zero address. if it is then we have a bug. we should try to mitigate overarching conditions

Reply:
I think this is a bug in the ENS resolver, but I haven't been able to track it down before the release. I've opened a separate PR to handle it #638